presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkDiskPageInput.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.execution.StageId;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkStorageHandle;
+import com.facebook.presto.spark.classloader_interface.PrestoSparkTaskOutput;
+import com.facebook.presto.spi.page.PagesSerde;
+import com.facebook.presto.spi.page.SerializedPage;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.spi.storage.TempDataOperationContext;
+import com.facebook.presto.spi.storage.TempStorage;
+import com.facebook.presto.spi.storage.TempStorageHandle;
+import io.airlift.slice.InputStreamSliceInput;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.zip.CRC32;
+
+import static com.facebook.presto.spi.page.PagesSerdeUtil.readSerializedPages;
+import static java.util.Collections.shuffle;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkDiskPageInput
+        implements PrestoSparkPageInput
+{
+    private static final Logger log = Logger.get(PrestoSparkDiskPageInput.class);
+
+    private final PagesSerde pagesSerde;
+    @GuardedBy("this")
+    private List<Iterator<Page>> pageIterators;
+    @GuardedBy("this")
+    private int currentIteratorIndex;
+    private long retainedSizeInBytes;
+    private final TempStorage tempStorage;
+    private final TempDataOperationContext tempDataOperationContext;
+    private final PrestoSparkBroadcastTableCacheManager prestoSparkBroadcastTableCacheManager;
+    private final StageId stageId;
+    private final PlanNodeId planNodeId;
+    private final List<List<PrestoSparkStorageHandle>> broadcastTableFilesInfo;
+    private final int storageBasedBroadcastJoinReadBufferSize;
+
+    public PrestoSparkDiskPageInput(
+            PagesSerde pagesSerde,
+            List<List<PrestoSparkStorageHandle>> broadcastTableFilesInfo,
```

Comment:
nit: move it after the `planNodeId ` to keep the fields and the parameters in the same order.